pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -345,8 +352,52 @@ func (am *Alertmanager) buildReceiverIntegrations(receiver *apimodels.PostableAp
 }
 
 // PutAlerts receives the alerts and then sends them through the corresponding route based on whenever the alert has a receiver embedded or not
-func (am *Alertmanager) PutAlerts(alerts apimodels.PostableAlerts) error {
-	return am.alerts.PutPostableAlert(alerts)
+func (am *Alertmanager) PutAlerts(postableAlerts apimodels.PostableAlerts) error {
+	now := time.Now()
+	alerts := make([]*types.Alert, 0, len(postableAlerts.PostableAlerts))
+	for _, a := range postableAlerts.PostableAlerts {
+		alerts = append(alerts, alertForDelivery(a, now))
+	}
+	return am.alerts.Put(alerts...)
+}
+
+func alertForDelivery(a amv2.PostableAlert, now time.Time) *types.Alert {
```

Comment:
I'm not sure this method or its name make much sense. In this context, `forDelivery` is not a concern of this system - this is just storing them in memory.

I'm in favour of just inlining it... for now. In the next PR, we can probably just move it to the a new file. Please let's make sure we add the remaining validations:

- Removal of empty Labels
- Must have a StartAt
- EndsAt must be before StartAt
- Validate label format
- Must have at least one label
- Validate annonnation format

Most of the above can be enforced by simply calling `alert.Validate()` and accumulating the errors.